server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -495,6 +501,14 @@ private static long jitter() {
     // add a random 10% wait
     return (long) ((1. + (r.nextDouble() / 10)) * TabletServer.TIME_BETWEEN_LOCATOR_CACHE_CLEARS);
   }
+  
+    //synchronized method to control simultaneous access
+  synchronized private Semaphore getSemaphore() {
+    if (sem == null) {
```

Comment:
to avoid some synchronization here, try something like this:
```
 private Semaphore getSemaphore() {
  if (sem == null) {
    synchronized(this) {
      if (sem == null) {
        sem = new Semaphore(maxThreadPermits);
      }
    }
    return sem;
  }
```